var-annotated error with imported BaseModel, related to namespace module

As a followup to https://github.com/typeddjango/django-stubs/issues/68 I've noticed that it would not work with:

```python
from django.db import models
from ...base.models import BaseModelMixin


# class BaseModelMixin(models.Model):
#     class Meta:
#         abstract = True


class TestModel(BaseModelMixin):
    foo = models.CharField(max_length=123)
```

`error: Need type annotation for "foo"  [var-annotated]`

The imported model is the same as the commented one.

If not importing it, but defining it there (uncommenting it), it works:

```python
from django.db import models
# from ...base.models import BaseModelMixin


class BaseModelMixin(models.Model):
    class Meta:
        abstract = True


class TestModel(BaseModelMixin):
    foo = models.CharField(max_length=123)
```

Note that it also fails when overwriting the import:
```python
from django.db import models
# from ...base.models import BaseModelMixin


class BaseModelMixin(models.Model):
    class Meta:
        abstract = True


class TestModel(BaseModelMixin):
    foo = models.CharField(max_length=123)
```

```
…/models.py:5: error: Name "BaseModelMixin" already defined (possibly by an import)  [no-redef]                                                                                                    
…/models.py:11: error: Need type annotation for "foo"  [var-annotated]                                                                                                                             
```

I've not investigated much yet, but it appears to be caused by the models fullname not containing the first part of the (namespaced) module name: `app.models.TestModel` vs `project.app.models.TestModel` (via `django_context.all_registered_model_class_fullnames`)..!

This results in the `is_model_subclass_info` check only working if it is defined in the same file (via `info.has_base` check - shouldn't that work better maybe in general?): https://github.com/typeddjango/django-stubs/blob/cf6952c9df385a699dd63054b6753eb924b0c68c/mypy_django_plugin/lib/helpers.py#L304-L307

Note that the namespace ("project") is added/used via `AppConfig`:

```python
class MyAppConfig(AppConfig):
    name = "project.app"
```

`project/__init__.py` does not exist, but creating it and/or changing `ProductConfig.name` to not include the namespace changes the behavior.

## System information

- OS: Arch Linux
- `python` version: 3.9.5
- `django` version: 4.0.dev20210602105309
- `mypy` version: 0.910
- `django-stubs` version: 1.8.0
- `django-stubs-ext` version: 0.2.0

--- a/mypy_django_plugin/django/context.py
+++ b/mypy_django_plugin/django/context.py
@@ -302,10 +302,6 @@ class DjangoContext:
         return all_model_bases
 
     @cached_property
-    def all_registered_model_class_fullnames(self) -> Set[str]:
-        return {helpers.get_class_fullname(cls) for cls in self.all_registered_model_classes}
-
-    @cached_property
     def model_class_fullnames_by_label(self) -> Mapping[str, str]:
         return {
             klass._meta.label: helpers.get_class_fullname(klass)

--- a/mypy_django_plugin/lib/helpers.py
+++ b/mypy_django_plugin/lib/helpers.py
@@ -401,12 +401,6 @@ def get_typechecker_api(ctx: Union[AttributeContext, MethodContext, FunctionCont
     return ctx.api
 
 
-def is_model_subclass_info(info: TypeInfo, django_context: "DjangoContext") -> bool:
-    return info.fullname in django_context.all_registered_model_class_fullnames or info.has_base(
-        fullnames.MODEL_CLASS_FULLNAME
-    )
-
-
 def check_types_compatible(
     ctx: Union[FunctionContext, MethodContext], *, expected_type: MypyType, actual_type: MypyType, error_message: str
 ) -> None:

--- a/mypy_django_plugin/main.py
+++ b/mypy_django_plugin/main.py
@@ -174,7 +174,7 @@ class NewSemanalDjangoPlugin(Plugin):
             if info.has_base(fullnames.FIELD_FULLNAME):
                 return partial(fields.transform_into_proper_return_type, django_context=self.django_context)
 
-            if helpers.is_model_subclass_info(info, self.django_context):
+            if helpers.is_model_type(info):
                 return partial(init_create.redefine_and_typecheck_model_init, django_context=self.django_context)
 
         return None

--- a/mypy_django_plugin/transformers/fields.py
+++ b/mypy_django_plugin/transformers/fields.py
@@ -24,7 +24,7 @@ def _get_current_field_from_assignment(
     ctx: FunctionContext, django_context: DjangoContext
 ) -> Optional[Union["Field[Any, Any]", ForeignObjectRel, "GenericForeignKey"]]:
     outer_model_info = helpers.get_typechecker_api(ctx).scope.active_class()
-    if outer_model_info is None or not helpers.is_model_subclass_info(outer_model_info, django_context):
+    if outer_model_info is None or not helpers.is_model_type(outer_model_info):
         return None
 
     field_name = None
@@ -234,7 +234,7 @@ def transform_into_proper_return_type(ctx: FunctionContext, django_context: Djan
     assert isinstance(default_return_type, Instance)
 
     outer_model_info = helpers.get_typechecker_api(ctx).scope.active_class()
-    if outer_model_info is None or not helpers.is_model_subclass_info(outer_model_info, django_context):
+    if outer_model_info is None or not helpers.is_model_type(outer_model_info):
         return ctx.default_return_type
 
     assert isinstance(outer_model_info, TypeInfo)

